src/query.rs

```diff
@@ -385,10 +385,15 @@ mod tests {
         // nondeterminism in how the results are ordered.
         let sort_individual_outputs = |results: &mut TestOutput| {
             let key_func = |elem: &BTreeMap<String, FieldValue>| {
-                (
-                    elem["span_filename"].as_str().unwrap().to_owned(),
-                    elem["span_begin_line"].as_usize().unwrap(),
-                )
+                let filename = elem
+                    .get("span_filename")
+                    .and_then(|value| value.as_str())
+                    .expect("a valid query must have span_filename");
+                let line = elem
+                    .get("span_begin_line")
+                    .expect("a valid query must have span_begin_line");
```

Comment:
Two minor suggestions:
- Can we check if _both_ values are missing, and if so then report them both missing at once?
- Can we make the `.expect()` text directly point the user to the relevant portion of the `CONTRIBUTING.md` document (perhaps even linking to it on GitHub?) so that they don't have to be confused about what the error means?

Reply:
Yup will do! About the linking to the relevant section, do we need to make some sort of heading somewhere so we can do `#heading_name` to link it?

Reply:
Certainly open to it if you think it's best, I trust your judgment. It could also be as simple as just linking to the file and mentioning that `span_filename` and `span_begin_line` should be output like that file describes, and we could improve it further from there if needed in the future.

Reply:
Sounds good